test/e2e/network/netpol/network_policy.go

```diff
@@ -571,6 +571,37 @@ var _ = common.SIGDescribe("Netpol", func() {
 			ValidateOrFail(k8s, &TestCase{ToPort: 81, Protocol: v1.ProtocolTCP, Reachability: reachabilityPort81})
 		})
 
+		ginkgo.It("should allow egress access from namespace on one named port while blocking other namespaces and ports[Feature:NetworkPolicy]", func(ctx context.Context) {
+			protocols := []v1.Protocol{protocolTCP}
+			ports := []int32{80, 81}
+			k8s = initializeResources(ctx, f, protocols, ports)
+			nsX, nsY, nsZ := getK8sNamespaces(k8s)
+			allowedLabels := &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					namespaceLabelKey: nsY,
+				},
+			}
+			egressRule := networkingv1.NetworkPolicyEgressRule{}
+			egressRule.Ports = append(egressRule.Ports, networkingv1.NetworkPolicyPort{Port: &intstr.IntOrString{Type: intstr.String, StrVal: "serve-80-tcp"}})
+			egressRule.To = append(egressRule.To, networkingv1.NetworkPolicyPeer{NamespaceSelector: allowedLabels})
+			policy := GenNetworkPolicyWithNameAndPodMatchLabel("allow-client-a-via-ns-selector-80", map[string]string{"pod": "a"}, SetSpecEgressRules(egressRule))
+			CreatePolicy(ctx, k8s, policy, nsX)
+
+			reachabilityPort80 := NewReachability(k8s.AllPodStrings(), true)
+			// disallow all traffic from the x or z namespaces
+			reachabilityPort80.ExpectPeer(&Peer{Namespace: nsX}, &Peer{Namespace: nsX, Pod: "a"}, false)
```

Comment:
get rid of this one, you arent blocking ingress into x/a, but rather, blocking egress out of a